libs/ledgerjs/packages/context-module/src/loaders/TokenLoader.ts

```diff
@@ -1,11 +1,44 @@
+import axios from "axios";
 import { LoaderOptions } from "../models/LoaderOptions";
 import { Transaction } from "../models/Transaction";
 import { ContextLoader } from "./ContextLoader";
 
-export class CalMetadataFetcher implements ContextLoader {
+export enum ERC20_SUPPORTED_SELECTORS {
+  Approve = "0x095ea7b3",
+  Transfer = "0xa9059cbb",
+}
+
+const SUPPORTED_SELECTORS: `0x${string}`[] = Object.values(ERC20_SUPPORTED_SELECTORS);
+
+export class TokenLoader implements ContextLoader {
   constructor() {}
 
-  load(_transaction: Transaction, _options: LoaderOptions) {
-    return Promise.resolve([]);
+  async load(transaction: Transaction, _options: LoaderOptions) {
+    if (!transaction.to || !transaction.data || transaction.data === "0x") {
+      return [];
+    }
+
+    const selector = transaction.data.slice(0, 10) as `0x${string}`;
+
+    if (!this.isSelectorSupported(selector)) {
+      return [];
+    }
+
+    const response = await axios.request<[{ live_signature: string }]>({
+      method: "GET",
+      url: `https://crypto-assets-service.api.ledger.com/v1/tokens`,
+      params: { search: transaction.to, chain_id: transaction.chainId },
```

Comment:
Hey :) `search` param is a fuzzy search, not cached and quite costly on backend side. If this is a contract address, you should use `contract_address` instead:
```suggestion
      params: { contract_address: transaction.to, chain_id: transaction.chainId },
```

Reply:
Also you will need `output: 'live_signature'`

Reply:
Will do, thx for the review 👍